File: jlox/lox/Token.java
package lox;

class Token {
    final TokenType type;
    final String lexeme;
    final Object literal;
    final int line;

    Token(TokenType type, String lexeme, Object literal, int line) {
        this.type = type;
        this.lexeme = lexeme;
        this.literal = literal;
        this.line = line;
    }

    public String toString() {
        return (
            String.format("%-15s", type) +
            String.format("%-10s", lexeme) +
            literal
        );
    }
}

File: golox/token.go
package main

import "fmt"

type Token struct {
	Type    TokenType
	Lexeme  string
	Literal interface{}
	Line    int
}

func (t Token) ToString() string {
	return fmt.Sprintf("%-15v %-10v %v", TokenTypeString(t.Type), t.Lexeme, t.Literal)
}
